sdk/dapp-kit/src/hooks/wallet/useSignAndExecuteTransactionBlock.ts

```diff
@@ -74,42 +75,83 @@ export function useSignAndExecuteTransactionBlock({
 				);
 			}
 
-			if (executeFromWallet) {
-				const walletFeature = currentWallet.features['sui:signAndExecuteTransactionBlock'];
+			const canExecuteFromWallet =
+				!!currentWallet.features['sui:signAndExecuteTransactionBlock:v2'];
+
+			const reportEffects =
+				currentWallet.features['sui:reportTransactionBlockEffects']
+					?.reportTransactionBlockEffects ?? (() => {});
+
+			if (canExecuteFromWallet) {
+				const walletFeature = currentWallet.features['sui:signAndExecuteTransactionBlock:v2'];
 				if (!walletFeature) {
 					throw new WalletFeatureNotSupportedError(
 						"This wallet doesn't support the `signAndExecuteTransactionBlock` feature.",
 					);
 				}
 
-				return walletFeature.signAndExecuteTransactionBlock({
+				const result = await walletFeature.signAndExecuteTransactionBlock({
 					...signTransactionBlockArgs,
+					transactionBlock: {
+						async toJSON() {
+							return typeof transactionBlock === 'string'
+								? transactionBlock
+								: await transactionBlock.toJSON({
+										supportedIntents,
+										client,
+								  });
+						},
+					},
 					account: signerAccount,
 					chain: signTransactionBlockArgs.chain ?? signerAccount.chains[0],
-					requestType,
-					options,
 				});
+
+				await reportEffects({ effects: result.effects });
+
+				return result;
 			}
 
-			const walletFeature = currentWallet.features['sui:signTransactionBlock'];
+			const walletFeature = currentWallet.features['sui:signTransactionBlock:v2'];
 			if (!walletFeature) {
 				throw new WalletFeatureNotSupportedError(
 					"This wallet doesn't support the `signTransactionBlock` feature.",
 				);
 			}
 
-			const { signature, transactionBlockBytes } = await walletFeature.signTransactionBlock({
+			const { signature, bytes } = await walletFeature.signTransactionBlock({
 				...signTransactionBlockArgs,
+				transactionBlock: {
+					async toJSON() {
+						return typeof transactionBlock === 'string'
+							? transactionBlock
+							: await transactionBlock.toJSON({
+									supportedIntents,
+									client,
+							  });
+					},
+				},
 				account: signerAccount,
 				chain: signTransactionBlockArgs.chain ?? signerAccount.chains[0],
 			});
 
-			return client.executeTransactionBlock({
-				transactionBlock: transactionBlockBytes,
+			const { rawEffects, digest } = await client.executeTransactionBlock({
+				transactionBlock: bytes,
 				signature,
-				requestType,
-				options,
+				options: {
+					showRawEffects: true,
+				},
 			});
+
+			const effects = toB64(new Uint8Array(rawEffects!));
```

Comment:
What is `rawEffects` returned from the API?

Reply:
An array of numbers